client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -243,10 +250,72 @@ class LogsStreamsService {
                 coldStorageNotifyEnabled: true,
                 coldStorageEnabled: false,
                 webSocketEnabled: true
-            }
+            },
+            loading: false
         };
     }
 
+    /**
+     * extracts graylog URL from stream. Shows error message on UI if no graylog URL is found.
+     *
+     * @param {any} stream
+     * @returns {string} graylog url, if not found empty string
+     * @memberof LogsStreamsService
+     */
+    getStreamGraylogUrl (stream) {
+        const url = this.UrlHelper.findUrl(stream, "GRAYLOG_WEBUI");
+        if (!url) {
+            this.CloudMessage.error(this.$translate.instant("logs_streams_get_graylog_url_error", { stream: stream.info.title }));
+        }
+        return url;
+    }
+
+    /**
+     * extracts and copies stream token to clipboard.
+     * Shows error message on UI if no no token found or browser does not support clipboard copy.
+     *
+     * @param {any} stream
+     * @memberof LogsStreamsService
+     */
+    copyStreamToken (stream) {
+        const token = this.getStreamToken(stream);
+        if (token) {
+            const error = this.ControllerHelper.copyToClipboard(token);
+            if (error) {
+                this.CloudMessage.error({ textHtml: this.$translate.instant("logs_streams_copy_token_error", {
```

Comment:
please us error handler / success handler as much as you can.  textHTML can be removed.